[Eng Wai Lam Simpson] iP

craeyeons opened the pull request:
# JeanPro
> Lots of tasks? Lots of worries? No worries! Jean is here just for you!

Jean frees your mind from the hassle of having to remember things!
- it is completely text-based
- very easy to use
- supports French! 🇫🇷 
- ~~fast to use~~ **_JEAN GOES BRRRRRRR_** 🚀 

To use it, simply,

1. download the `jean.jar` file from [here](https://github.com/craeyeons/ip/releases).
2. double click on it
3. add your tasks in
4. and you are now at least 1 worry free!

And it is all **FREE**!

# Features 
- [x] Managing Tasks!
- [x] Managing Deadlines!
- [ ] Reminders! (coming soon!)

---
If you're a Java Programmer, you can use it to practice your Java too. Here's the `main` method:
```java
public static void main(String[] args) {
        Ui ui = new Ui();
        Storage storage = new Storage("data/list.txt");
        TaskList tasks = null;
        try {
            File dir = new File("data");
            if (!dir.exists()) {
                dir.mkdirs();
            }
            tasks = new TaskList(storage.load());
        } catch (JeanException e) {
            ui.showJeanError(e.getMessage());
            tasks = new TaskList();
        } catch (IOException e) {
            ui.showGeneralError(e.getMessage());
        }
        new Jean(ui, storage, tasks).run();
    }
```

Commit:
Add Gradle support

craeyeons committed:
completed Level-1

craeyeons committed:
completed Level-2

craeyeons committed:
fixed typo

craeyeons committed:
added Task.java class to treat classes as an object

craeyeons committed:
completed Level-3

craeyeons committed:
fixed indentation

craeyeons committed:
completed Level-4

craeyeons committed:
added testfiles, input.txt and expected.txt

craeyeons committed:
fixed indentation

craeyeons committed:
fixed I/O, indentation and testcases

craeyeons committed:
fixed: parse string outside of function

craeyeons committed:
renamed Duke to Jean

craeyeons committed:
completed Level-5

craeyeons committed:
access of class variable numberOfTasks using class name instead of an instance

craeyeons committed:
added the case if the command does not exist

craeyeons committed:
completed Level-6

craeyeons committed:
grouped try/catch clauses for each operation to its respective check function to reduce clutter

craeyeons committed:
fixed text feedback formatting. added french ver. for exceptions.

craeyeons committed:
updated exception conditions

craeyeons committed:
added exception cases and updated test files

craeyeons committed:
Complete Level-7

craeyeons committed:
Complete Level-8

craeyeons committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Complete Level-7

craeyeons committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Complete Level-8

# Conflicts:
#	src/main/java/Jean.java
Retain only used packages

craeyeons committed:
Complete A-MoreOOP

craeyeons committed:
Complete A-Packages

craeyeons committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

* 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip:
  Add Gradle support

craeyeons committed:
Change Jean constructor and main method to comply with dependency injection.

craeyeons committed:
Complete A-JUnit

craeyeons committed:
Fix saveFile method in Storage.

craeyeons committed:
Complete A-Jar

craeyeons committed:
Complete A-JavaDoc

craeyeons committed:
Complete A-CodingStandard

craeyeons committed:
Complete Level-9

craeyeons committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Complete A-JavaDoc

craeyeons committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Complete Level-9

# Conflicts:
#	src/main/java/jean/parser/Parser.java
#	src/main/java/jean/task/TaskList.java

craeyeons committed:
Merge branch 'branch-A-CodingStandards'

* branch-A-CodingStandards:
  Complete A-CodingStandard

# Conflicts:
#	src/main/java/jean/Jean.java
#	src/main/java/jean/command/Command.java
#	src/main/java/jean/command/DeleteCommand.java
#	src/main/java/jean/command/ExitCommand.java
#	src/main/java/jean/command/MarkCommand.java
#	src/main/java/jean/command/UnmarkCommand.java
#	src/main/java/jean/exception/JeanException.java
#	src/main/java/jean/parser/Parser.java
#	src/main/java/jean/storage/Storage.java

craeyeons committed:
Add JavaDocs for a few methods and Remove unnecessary imports.

craeyeons committed:
Ignore data/list.txt

craeyeons committed:
Remove data/list.txt from remote repository

craeyeons committed:
Complete Level-10

craeyeons committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Complete Level-10

craeyeons committed:
Complete A-CheckStyle.

craeyeons committed:
Add title to ui.

ErvinK123 commented on src/main/java/jean/Jean.java:
```diff
@@ -0,0 +1,84 @@
+package jean;
```
Good organisation of package 👍

ErvinK123 commented on src/main/java/jean/command/AddCommand.java:
```diff
@@ -0,0 +1,35 @@
+/**
+ * Represents a Command which instructs Jean to add a task into the list of tasks.
+ */
+public class AddCommand extends Command {
```
Good use of inheritance 👍

ErvinK123 reviewed (approved):
LGTM , overall coding standard looks clean

liviamil commented on src/main/java/jean/parser/Parser.java:
```diff
@@ -0,0 +1,150 @@
+package jean.parser;
```
I like the control flow, very easy to follow!

liviamil commented on src/main/java/jean/parser/Parser.java:
```diff
@@ -0,0 +1,150 @@
+        }
+    }
+
+    private static FindCommand checkFind(String fullCommand) throws JeanException {
```
Some JavaDoc comments seems to be missing?

liviamil reviewed (commented):
Looks good! Organized well and easy to follow.

ytingtan reviewed (commented):
I enjoyed reading your code! It was very nicely structured and code quality is very good. Most notably, I like your package organisation! Makes everything very neat and easy to read. Also liked the French bits in the bot. 😄 Overall LGTM!

craeyeons committed:
Jean has no assertions: add assertions to Jean.

There are certain situations where the program should throw an exception instead of proceeding. (e.g. empty command input -> should throw JeanException, not return an invalidCommand)

Add assertions at appropriate points of the code to check and avoid such scenarios.

Using assertions as it allows us to separate testing from the final
product itself easily.

craeyeons committed:
Direct comparison of command lines with strings of valid command: make
comparison safer with constant strings.

There are only few valid commands in Jean. It should be verified via constant strings to reduce possible errors.

Let's extract all valid command lines as constants.

craeyeons committed:
Merge pull request #1 from craeyeons/branch-A-Assertions

Jean has no assertions: add assertions to Jean.

craeyeons committed:
Merge branch 'master' of https://github.com/craeyeons/ip

* 'master' of https://github.com/craeyeons/ip:
  Jean has no assertions: add assertions to Jean.

craeyeons committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Jean has no assertions: add assertions to Jean.

# Conflicts:
#	src/main/java/jean/parser/Parser.java

craeyeons committed:
Merge pull request #3 from craeyeons/branch-A-CodeQuality

Branch A-CodeQuality

craeyeons committed:
Merge branch 'master' of https://github.com/craeyeons/ip

* 'master' of https://github.com/craeyeons/ip:
  Direct comparison of command lines with strings of valid command: make comparison safer with constant strings.

craeyeons committed:
Sort command: implement sort command in Jean.

Sort command did not exist to sort commands in alphabetical order.

A task list in alphabetical order would allow the user to easier find
the task he/she wants to mark. With this the user does not need to
delete -> then reinsert such tasks into the list.

Let's
1. create a SortCommand class
2. let Task Class implement Comparator to allow the tasks to be sorted
via their descriptions in alphabetical order.

craeyeons committed:
Merge branch 'branch-BCD-Extension'

* branch-BCD-Extension:
  Sort command: implement sort command in Jean.

craeyeons committed:
Complete A-UserGuide

craeyeons committed:
Fix user guide.

craeyeons committed:
Update build.gradle file.

craeyeons committed:
Fix build.gradle.

craeyeons committed:
Remove Ui.png.

craeyeons committed:
Add Ui.png.
